pennylane/beta/tapes/tape.py

```diff
@@ -936,3 +936,387 @@ def execute_device(self, params, device):
     # if they need to perform any logic in between the user's
     # call to tape.execute and the internal call to tape.execute_device.
     _execute = execute_device
+
+    # ========================================================
+    # gradient methods
+    # ========================================================
+
+    def _grad_method(self, idx, use_graph=True, default_method="F"):
+        """Determine the correct partial derivative computation method for each gate parameter.
+
+        Parameter gradient methods include:
+
+        * ``None``: the parameter does not support differentiation.
+
+        * ``"0"``: the variational circuit output does not depend on this
+            parameter (the partial derivative is zero).
+
+        * ``"F"``: the parameter has a non-zero derivative that should be computed
+            using finite-differences.
+
+        * ``"A"``: the parameter has a non-zero derivative that should be computed
+            using an analytic method.
+
+        .. note::
+
+            The base ``QuantumTape`` class only supports numerical differentiation, so
+            this method will always return either ``"F"`` or ``None``. If an inheriting
+            tape supports analytic differentiation for certain operations, make sure
+            that this method is overwritten appropriately to return ``"A"`` where
+            required.
+
+        Args:
+            idx (int): parameter index
+            use_graph: whether to use a directed-acyclic graph to determine
+                if the parameter has a gradient of 0
+            default_method (str): the default differentiation value to return
+                for parameters that where the grad method is not ``"0"`` or ``None``
+
+        Returns:
+            str: partial derivative method to be used
+        """
+        op = self._par_info[idx]["op"]
+
+        if op.grad_method is None:
+            return None
```

Comment:
What would that mean?

Reply:
Yep, currently `op.grad_method = None` means a non-differentiable parameter (for example, `QubitUnitary`).